builtin/providers/kubernetes/resource_kubernetes_resource_quota.go

```diff
@@ -0,0 +1,211 @@
+							Optional:     true,
+							Elem:         schema.TypeString,
+							ValidateFunc: validateResourceList,
+						},
+						"scopes": {
+							Type:        schema.TypeSet,
+							Description: "A collection of filters that must match each object tracked by a quota. If not specified, the quota matches all objects.",
+							Optional:    true,
+							ForceNew:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Set:         schema.HashString,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceKubernetesResourceQuotaCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	spec, err := expandResourceQuotaSpec(d.Get("spec").([]interface{}))
+	if err != nil {
+		return err
+	}
+	resQuota := api.ResourceQuota{
+		ObjectMeta: metadata,
+		Spec:       spec,
+	}
+	log.Printf("[INFO] Creating new resource quota: %#v", resQuota)
+	out, err := conn.CoreV1().ResourceQuotas(metadata.Namespace).Create(&resQuota)
+	if err != nil {
+		return fmt.Errorf("Failed to create resource quota: %s", err)
+	}
+	log.Printf("[INFO] Submitted new resource quota: %#v", out)
+	d.SetId(buildId(out.ObjectMeta))
+
+	err = resource.Retry(1*time.Minute, func() *resource.RetryError {
+		quota, err := conn.CoreV1().ResourceQuotas(out.Namespace).Get(out.Name)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		if resourceListEquals(spec.Hard, quota.Status.Hard) {
+			return nil
+		}
+		err = fmt.Errorf("Quotas don't match after creation.\nExpected: %#v\nGiven: %#v",
+			spec.Hard, quota.Status.Hard)
+		return resource.RetryableError(err)
+	})
+	if err != nil {
+		return err
+	}
+
+	return resourceKubernetesResourceQuotaRead(d, meta)
+}
+
+func resourceKubernetesResourceQuotaRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	namespace, name := idParts(d.Id())
+	log.Printf("[INFO] Reading resource quota %s", name)
+	resQuota, err := conn.CoreV1().ResourceQuotas(namespace).Get(name)
+	if err != nil {
+		log.Printf("[DEBUG] Received error: %#v", err)
+		return err
+	}
+	log.Printf("[INFO] Received resource quota: %#v", resQuota)
+
+	// This is to work around K8S bug
+	// See https://github.com/kubernetes/kubernetes/issues/44539
+	if resQuota.ObjectMeta.GenerateName == "" {
+		if v, ok := d.GetOk("metadata.0.generate_name"); ok {
+			resQuota.ObjectMeta.GenerateName = v.(string)
+		}
+	}
+
+	err = d.Set("metadata", flattenMetadata(resQuota.ObjectMeta))
+	if err != nil {
+		return err
+	}
+	err = d.Set("spec", flattenResourceQuotaSpec(resQuota.Spec))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceKubernetesResourceQuotaUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
```

Comment:
`ops := patchMetadata("metadata.0.", "/metadata/", d)` should return empty slice of `PatchOperations` if it has no changes and if we have reached the `Update()` function it's clear that we have _something_ to update - either `spec` or `metadata`, so it is there just to initialize the slice, really.

I could change it to something like

```go
ops := make([]PatchOperations, 0, 0)
if d.HasChange("metadata") {
    ops = append(ops, patchMetadata("metadata.0.", "/metadata/", d)
}
```
if you think it's better?